themes/default/content/blog/next-level-iac-package-ecosystems/index.md

```diff
@@ -60,26 +64,95 @@ export const bucketName = bucket.id;
-``` 
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import pulumi
+import pulumi_aws as aws
+
+# Create an S3 bucket
+bucket = aws.s3.Bucket('my-bucket')
+
+# The path to the file to upload to S3
+file_path = "path/to/your/local/file.ext"  # Replace with the path to the file you want to upload
+
+# Read the contents of the file to be uploaded to S3.
+with open(file_path, 'rb') as file:
+    file_contents = file.read()
+
+# Upload the file to the S3 bucket
+bucket_object = aws.s3.BucketObject('my-bucket-object',
+                                bucket=bucket.id, # Reference to the bucket created above.
+                                key='file.ext',  # This will be the file's name in the bucket.
+                                content=file_contents,
+                                content_type='text/plain')  # MIME-type of the file. Adjust if you upload a different type of file.
+
+# Export the name of the bucket
+pulumi.export('bucket_name', bucket.id)
+
+# Export the URL of the file in the S3 bucket
+pulumi.export('object_url', pulumi.Output.concat('s3://', bucket.id, '/', bucket_object.key))
+```
+
+{{% /choosable %}}
+
 
 However, something that stands out as a weak link here is the line:
 
+{{< chooser language "typescript,python" />}}
+
+{{% choosable language typescript %}}
+
 ```typescript
 contentType: "text/plain", // MIME-type of the file. Adjust if you upload a different type of file.
 ```
 
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+content_type='text/plain')  # MIME-type of the file. Adjust if you upload a different type of file.
+```
+{{% /choosable %}}
+
 Yeah, hard-coding the MIME type for every file I need to upload is a not going to scale. In fact, I’m probably going to get a random directory full of files I need to iterate over, each with different file types, and it will change every time it runs, and we won’t know what type of file it is ahead of time.
 
-Packages to the rescue!! Turns out that determining the MIME type of a file in TypeScript only involves importing the community-created [`mime-types`][npm-mimetypes] library and a single function call to `mime.lookup(...)`:
+Packages to the rescue!! Turns out that determining the MIME type of a file in TypeScript only involves importing a community-created [`mime-types`][npm-mimetypes] [library][python-mimetypes] and a single function call to guess the mimetype:
+
+{{< chooser language "typescript,python" />}}
+
+{{% choosable language typescript %}}
 
 ```typescript
 import * as mime from 'mime-types'
 
-const mimeType = mime.lookup(‘path/to/your/local/file.ext’);
+// Detect the MIME type of the file
+const mimeType = mime.lookup('path/to/your/local/file.ext');
+
+```
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import mimetypes
 
+# Detect the MIME type of the file
+mime_type, _ = mimetypes.guess_type('path/to/your/local/file.ext')
```

Comment:
Hm -- does this actually give the expected MIME type? It seems to give me `application/vnd.novadigm.ext`. Should we not just stick with `.txt`?